src/dataDisplay/tag/tag.jsx

```diff
@@ -0,0 +1,339 @@
+                    boxShadow: `inset 1px 0 0 ${palette.success.dark}`,
+                    '&:hover': {
+                        backgroundColor: palette.success.dark,
+                    },
+                },
+                '&transparent .label-clearable-button': {
+                    backgroundColor: 'transparent',
+                    boxShadow: 'none',
+                    color: palette.text.secondary,
+                },
+                '&warning .label-clearable-button': {
+                    backgroundColor: palette.warning.dark,
+                    '&:hover': {
+                        boxShadow: `inset 1px 0 0 ${palette.warning.dark}`,
+                    },
+                },
+            },
+        },
+        '&.label-clickable': {
+            cursor: 'pointer',
+        },
+        '&.label-fluid': {
+            width: '100%',
+        },
+        '& .label-inner-container': {
+            width: '100%',
+        },
+    },
+});
+
+class Label extends React.Component {
+    constructor() {
+        super();
+
+        this.onClearClick = this.onClearClick.bind(this);
+        this.onClearKeyDown = this.onClearKeyDown.bind(this);
+        this.onClick = this.onClick.bind(this);
+        this.onKeyDown = this.onKeyDown.bind(this);
+    }
+
+    onClearClick() {
+        const {
+            onClearClick,
+        } = this.props;
+
+        if (isFunction(onClearClick)) {
+            onClearClick();
+        }
+    }
+
+    onClearKeyDown() {
+        const {
+            onClearKeyDown,
+        } = this.props;
+
+        if (isFunction(onClearKeyDown)) {
+            onClearKeyDown();
+        }
+    }
+
+    onClick() {
+        const {
+            onClick,
+        } = this.props;
+
+        if (isFunction(onClick)) {
+            onClick();
+        }
+    }
+
+    onKeyDown() {
+        const {
+            onKeyDown,
+        } = this.props;
+
+        if (isFunction(onKeyDown)) {
+            onKeyDown();
+        }
+    }
+
+    render() {
+        const {
+            children,
+            classes,
+            className,
+            color,
+            fluid,
+            inverse,
+            onClick,
+            onClearClick,
+            style,
+            tabIndex,
+        } = this.props;
+
+        const rootClasses = ClassNames(
+            UI_CLASS_NAME,
+            'label',
+            classes.root,
+            className,
+            {
+                'label-clickable': onClick,
+                'label-color-alert': color === 'alert',
+                'label-color-highlight': color === 'highlight',
+                'label-color-primary': color === 'primary',
+                'label-color-success': color === 'success',
+                'label-color-transparent': color === 'transparent',
+                'label-color-warning': color === 'warning',
+                'label-fluid': fluid,
+                'label-inverse': inverse,
+                'label-clearable': onClearClick,
```

Comment:
Same ... these should all be `'tag-*` not `'label-*`, right?  Unless we wanted to leave it deliberately?

Reply:
Fixed.

Reply:
Looks good.  Resolving.